File: examples/python/async/create_raw_iterator.py
from BinaryOptionsToolsV2.pocketoption import PocketOptionAsync
from BinaryOptionsToolsV2.validator import Validator
from datetime import timedelta

import asyncio


async def main(ssid: str):
    # Initialize the API client
    api = PocketOptionAsync(ssid)
    await asyncio.sleep(5)  # Wait for connection to establish

    # Create a validator for price updates
    validator = Validator.regex(r'{"price":\d+\.\d+}')

    # Create an iterator with 5 minute timeout
    stream = await api.create_raw_iterator(
        '42["price/subscribe"]',  # WebSocket subscription message
        validator,
        timeout=timedelta(minutes=5),
    )

    try:
        # Process messages as they arrive
        async for message in stream:
            print(f"Received message: {message}")
    except TimeoutError:
        print("Stream timed out after 5 minutes")
    except Exception as e:
        print(f"Error processing stream: {e}")


if __name__ == "__main__":
    ssid = input("Please enter your ssid: ")
    asyncio.run(main(ssid))

File: examples/javascript/create_raw_iterator.js
const { PocketOption } = require("./binary-options-tools.node");
const { Validator } = require("./binary-options-tools.node");

async function main(ssid) {
  // Initialize the API client
  const api = new PocketOption(ssid);

  // Wait for connection to establish
  await new Promise((resolve) => setTimeout(resolve, 5000));

  // Create a validator for price updates
  const validator = Validator.regex(/{"price":\d+\.\d+}/);

  try {
    // Create an iterator with 5 minute timeout
    const stream = api.createRawIterator(
      '42["price/subscribe"]', // WebSocket subscription message
      validator,
      { timeout: 5 * 60 * 1000 }, // 5 minutes in milliseconds
    );

    // Process messages as they arrive
    for await (const message of stream) {
      console.log(`Received message: ${message}`);
    }
  } catch (error) {
    if (error.name === "TimeoutError") {
      console.log("Stream timed out after 5 minutes");
    } else {
      console.log(`Error processing stream: ${error}`);
    }
  }
}

const ssid = "";

main(ssid).catch(console.error);
